packages/circuit-ui/components/TopNavigation/TopNavigation.tsx

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2021, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactNode } from 'react';
+import { css } from '@emotion/core';
+import { Theme } from '@sumup/design-tokens';
+
+import styled, { StyleProps } from '../../styles/styled';
+import { focusVisible } from '../../styles/style-mixins';
+import Hamburger, { HamburgerProps } from '../Hamburger';
+
+import { ProfileMenu, ProfileMenuProps } from './components/ProfileMenu';
+import { UtilityLinks, UtilityLinksProps } from './components/UtilityLinks';
+
+const headerStyles = ({ theme }: StyleProps) => css`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  width: 100%;
+  height: 49px; /* height + border-bottom */
+  background-color: ${theme.colors.bodyBg};
+  border-bottom: ${theme.borderWidth.kilo} solid ${theme.colors.n200};
+`;
+
+const Header = styled.header(headerStyles);
+
+const hamburgerStyles = (theme: Theme) => css`
+  ${focusVisible('inset')(theme)};
+
+  border-radius: 0;
+  border-right: ${theme.borderWidth.kilo} solid ${theme.colors.n200} !important;
+
+  ${theme.mq.mega} {
+    display: none;
+  }
+`;
+
+const logoStyles = ({ theme }: StyleProps) => css`
+  height: ${theme.iconSizes.tera};
+
+  > * {
+    display: block;
+    height: inherit;
+    line-height: 0;
+    padding: ${theme.spacings.kilo};
+  }
```

Comment:
Do we need this in here or should it be styled in whichever component is passed as the logo?

Reply:
Because it seems to me that in this case, we might as well just hardcode the SumUp logo?

Reply:
I chose not to hard-code the logo to leave it flexible (e.g. for Shoplo or a SumUp sub-brand). This also allows for shenanigans like displaying the page title next to the logo in the dashboard 👀 

Those base styles are useful to size and scale the logo appropriately. It's inspired by the approach that Storybook takes for custom logos. I don't feel too strongly about it, if you think we should remove those styles, I'd be happy to do that.

Reply:
I think that's fine, thanks for the rationale 👍 

🙈 the shenanigans